src/frontend/state/AchievementState.ts

```diff
@@ -0,0 +1,312 @@
+import { makeAutoObservable, runInAction } from 'mobx'
+import walletState from 'frontend/state/WalletState'
+import type {
+  Achievement,
+  AchievementFilter,
+  AchievementSort,
+  AchievementStore,
+  AchievementsStats,
+  SummaryAchievement
+} from 'common/types'
+
+interface AchievementData {
+  data: SummaryAchievement[]
+  pagesFetched: number
+  totalPages: number
+}
+
+interface AchievementRequestKey {
+  filter: AchievementFilter
+  sort: AchievementSort
+  store: AchievementStore
+}
+
+interface AchievementSortOption {
+  text: string
+  value: AchievementSort
+}
+
+interface IndividualAchievementKey {
+  gameId: string
+  sort: AchievementSort
+  store: AchievementStore
+}
+
+interface IndividualAchievementData {
+  data: Achievement[]
+}
+
+export const ACHIEVEMENT_SORT_OPTIONS = [
+  { text: 'Alphabetically (ASC)', value: 'ALPHA_A_TO_Z' },
+  { text: 'Alphabetically (DES)', value: 'ALPHA_Z_TO_A' }
+] as AchievementSortOption[]
+
+class AchievementState {
+  /**
+   * feature flag for enabled stores
+   */
+  storesEnabled = ['STEAM']
+
+  store = 'STEAM' as AchievementStore
+  playerStoreId = ''
+
+  // Stats
+  newAchievements = 0
+  totalAchievements = 0
+  totalGames = 0
+  mintedAchievements = 0
+  numFreeMints = 0
+
+  // Summary Achievements
+  summaryAchievements = new Map<string, AchievementData>()
+  currentFilter: AchievementFilter = 'ALL'
+  currentSort: AchievementSortOption = ACHIEVEMENT_SORT_OPTIONS[0]
+  currentStore: AchievementStore = 'STEAM'
+
+  // Individual Achievements
+  individualAchievements = new Map<string, IndividualAchievementData>()
+  currentIndividualSort: AchievementSortOption = ACHIEVEMENT_SORT_OPTIONS[0]
+
+  // Achievements Feature Flag
+  showAchievements = false
+
+  fetching = false
+
+  syncing = false
+
+  constructor() {
+    makeAutoObservable(
+      this,
+      {},
+      {
+        deep: true,
+        proxy: false,
+        name: 'AchievementState'
+      }
+    )
+  }
+
+  init() {
+    window.api
+      .shouldShowAchievements()
+      .then((res) => (this.showAchievements = res))
+  }
+
+  getAchievementsStats = () => {
+    window.api
+      .getAchievementsStats({
+        store: this.store,
+        playerStoreId: this.playerStoreId,
+        playerAddress: walletState.address
+      })
+      .then(this.setStats.bind(this))
```

Comment:
How about the error handling, how is being handled here? Same for the `shouldShowAchievements()`

Maybe something like so 

```typescript
  init = async () => {
    try {
      const res = await window.api.shouldShowAchievements();
      runInAction(() => {
        this.showAchievements = res;
      });
    } catch (error) {
      console.error('Error initializing achievements:', error);
    }
  }

  getAchievementsStats = async () => {
    try {
      const stats = await window.api.getAchievementsStats({
        store: this.store,
        playerStoreId: this.playerStoreId,
        playerAddress: walletState.address
      });
      this.setStats(stats);
    } catch (error) {
      console.error('Error fetching achievements stats:', error);
    }
  }
````

Reply:
I think a console error is fine here. the behavior will be that these values will remain 0

```
    this.newAchievements = state.newAchievements
    this.totalAchievements = state.totalAchievements
    this.totalGames = state.totalGames
    this.mintedAchievements = state.mintedAchievements
    this.numFreeMints = state.numFreeMints
```